### ISSUE TITLE
Example repo or walkthrough of a custom RxVirtualScrollStrategy implementation?

### ISSUE THREAD

**Issue by drakedeatonuk** [rx-angular/rx-angular#1613]
# Problem Solved By The Feature

Love the virtual scroller. Noticeably more performant than the cdk equivalent. However in my cdk scroller implementation I'm using a custom `VirtualScrollStrategy` which I wrote myself. To make the switch to rx-angular's virtual scroller fully, I need to implement a custom `RxVirtualScrollStrategy`, but the process on how to do this isn't clear enough for me. I can't even find an example of an implementation to review, and whereas with the cdk scroller there's many examples of how to implement a custom scroll strategy, for rx-angular there isn't any. This makes it hard for all but the most experienced of developers to adopt this tech.

# Solution

What would be useful would be either: 
1. a code repository with an example custom `RxVirtualScrollStrategy` implementation; or even better
2. a step-by-step walkthrough on how to create one.

**Comment by hoebbelsB** [comment]
hey @drakedeatonuk thank u for the input!

May I ask what the specific needs are that force you to implement a custom scroll strategy? For me this was the hardest task to implement, so I'm not recommending doing so if the already existing behavior fits your needs.

Anyway, we have at least a tiny section in the [docs](https://www.rx-angular.io/docs/template/api/virtual-scrolling#custom-scrollstrategies) explaining on how to create the skeleton directive. 
You have to implement the [`RxVirtualScrollStrategy` interface](https://github.com/rx-angular/rx-angular/blob/084f4fa1f503054d9efb714b980f08f55530b09b/libs/template/experimental/virtual-scrolling/src/lib/model.ts#L53C23-L53C46).

If you want to see how this interface is used, you can just go ahead and take a look at the ones that are provided by our library. You find them in [this folder](https://github.com/rx-angular/rx-angular/tree/main/libs/template/experimental/virtual-scrolling/src/lib/scroll-strategies). I recommend starting with the `fixed-size` strategy, as this is the simplest one to understand.

Sorry for not having specific docs about that, your request is very valid and I try to find some time to create a guide on how to. I guess the API isn't polished enough yet, things might change in future. Remember, it's still "experimental".

**Comment by drakedeatonuk** [comment]
Hi @hoebbelsB - big fan of your work!

In terms of usecase - I've built a chat application using the RxVirtualScrollViewportComponent. The reason I was investigating a custom scroll strategy was that I was getting some unexpected behaviour when prepending the rendered `RxVirtualScrollViewportComponent` array with additional chat replies from a database. On the chat app's user interface, older replies are rendered "above" newer ones, so they must be prepended to the start of the array in order to appear "above" the newer chat replies at the end of the array, rendered at the bottom of the ui. Any time the array was prepended, the virtual scroller ui would "skip" in a way I couldn't control. Additionally, when the array is prepended & the virtual scroll ui immediately "skips", the `scrollToIndex` function only works after a short delay of half a second or so - not sure why. This meant that there was no way to prepend the array without a visual ui "skip" that the user would see.  Anyways, because of this issue, I was investigating how a custom scroll strategy might give me the control I needed to better control the skipping that was occurring.

I'm currently using the `DynamicSizeVirtualScrollStrategy`, which solves all other challenges I've faced,  but the above problem remains.

Thanks for the links to the the existing implementations! I hadn't found those. Very useful & much appreciated 🙏

**Comment by hoebbelsB** [comment]
You are right, the scrollstrategy would provide you the level of control you need. But this sounds like a common use case and it should be possible to achieve a behavior like this with the already existing APIs. Is it somehow possible you can provide a bit of example code showcasing the issue? I could try to find a solution with the existing scroll strategy.
If appending data is causing issues with the current implementation, I anyway need to find a fix for that.
